examples/set-new-validators/index.ts

```diff
@@ -0,0 +1,115 @@
+  return chain.blockExplorers?.default.url;
+}
+
+// set the parent chain and create a public client for it
+const parentChain = arbitrumSepolia;
+const parentChainPublicClient = createPublicClient({
+  chain: parentChain,
+  transport: http(),
+}).extend(
+  rollupAdminLogicPublicActions({
+    rollup: process.env.ROLLUP_ADDRESS as `0x${string}`,
+  }),
+);
+
+// load the deployer account
+const deployer = privateKeyToAccount(sanitizePrivateKey(process.env.ROLLUP_OWNER_PRIVATE_KEY));
+
+async function main() {
+  //
+  // Parent chain
+  // ------------
+  //
+  // get the core contracts
+  console.log('Fetching UpgradeExecutor address in the parent chain...');
+  const transactionHash = await createRollupFetchTransactionHash({
+    rollup: process.env.ROLLUP_ADDRESS as `0x${string}`,
+    publicClient: parentChainPublicClient,
+  });
+  const transactionReceipt = createRollupPrepareTransactionReceipt(
+    await parentChainPublicClient.getTransactionReceipt({ hash: transactionHash }),
+  );
+  const coreContracts = transactionReceipt.getCoreContracts();
+  console.log(`UpgradeExecutor address: ${coreContracts.upgradeExecutor}`);
+
+  // here we just set one new validator, so the array length is 1
+  const newValidators = [process.env.NEW_VALIDATOR_ADDRESS as `0x${string}`];
+  const newValidatorStatus = [true];
+
+  // check the status of this address in validator list before executing
+  const beforeStatus = await parentChainPublicClient.rollupAdminLogicReadContract({
+    functionName: 'isValidator',
+    args: [newValidators[0]],
+    rollup: coreContracts.rollup,
+  });
+  console.log(
+    `Before executing, the address ${newValidators[0]} status in validator list is ${beforeStatus}`,
+  );
+
+  // prepare set validator transaction request
+  const setValidatorTransactionRequest =
+    await parentChainPublicClient.rollupAdminLogicPrepareTransactionRequest({
+      functionName: 'setValidator',
+      args: [
+        newValidators, // validator address list
+        newValidatorStatus, // validator status list
+      ],
+      upgradeExecutor: coreContracts.upgradeExecutor,
+      account: deployer.address,
+      rollup: coreContracts.rollup,
+    });
+
+  // sign and send the transaction
+  const setValidatorTransactionHash = await parentChainPublicClient.sendRawTransaction({
+    serializedTransaction: await deployer.signTransaction(setValidatorTransactionRequest),
+  });
+
+  // wait for transaction receipt
+  const setValidatorTransactionReceipt = await parentChainPublicClient.waitForTransactionReceipt({
+    hash: setValidatorTransactionHash,
+  });
+
+  console.log(
+    `New validator address set in ${getBlockExplorerUrl(parentChain)}/tx/${
+      setValidatorTransactionReceipt.transactionHash
+    }`,
+  );
+
+  // Check the status of this address in validator list before executing
+  const afterStatus = await parentChainPublicClient.rollupAdminLogicReadContract({
+    functionName: 'isValidator',
+    args: [newValidators[0]],
```

Comment:
NIT:
```suggestion
    args: newValidators
```

Reply:
i actually think that `[newValidators[0]]` is clearer than `newValidators` for its purpose here

Reply:
I found this is how we handle it on `src/decorators/rollupAdminLogicPublicActions.integration.test.ts` so I used the same way.